extensions/mssql/src/controllers/mainController.ts

```diff
@@ -628,11 +629,76 @@ export default class MainController implements vscode.Disposable {
             );
 
             this.registerLanguageModelTools();
+            this.registerTaskCompletionHandlers();
 
             return true;
         }
     }
 
+    /**
+     * Helper method to register task completion handlers for DacPac operations
+     */
+    private registerTaskCompletionHandlers(): void {
```

Comment:
If this method is only used for DACPAC-related completion handlers, could we rename it to something more specific like registerDacpacTaskCompletionHandlers?

Also, I’m wondering if we should start moving some of this logic out of mainController into feature-specific modules (e.g. a DACPAC-specific file) and just import/register them here. This file is starting to get quite large, and splitting by feature area might make it easier to maintain.